### PR TITLE
[FIX] microsoft_calendar: remove check on create

### PR DESCRIPTION
**Pull request by ryce-odoo**
Added condition on check that ensured the organizer was an attendee on the event on
create. This was added in a change that allowed the organizer to be
changed on the Odoo side. However, this caused issues with the
appointments app when creating an appointment that only used resources.
The "organizer" who in this case is the creator of the appointment type
would not be an attendee and therefore would cause an error and make the
appointment type unbookable.

Adding this condition allows the appointment to bypass the check if the
appointment type uses resources instead of users.

opw-3841495
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by ryce-odoo** [comment]
Hey @geflx 
I actually like that idea a lot and had thought about something similar. I have never really done a multi module PR before especially across repos so I didn't consider this option. Would I have to make a separate PR in enterprise and link back here?

**Comment by ryce-odoo** [comment]
Hey @geflx 
I have made the changes accordingly and made the pr to enterprise as well here:
https://github.com/odoo/enterprise/pull/61731

File: addons/microsoft_calendar/models/calendar.py
Comment (reth-odoo) [inline addons/microsoft_calendar/models/calendar.py:115]: Careful that this is `microsoft_calendar`, appointment doesn't depend on this module so you should add the method in base calendar.

Also, your override cannot take in more/different arguments than the original method, as in regular inheritance. Here I would just pass in "vals" and let the method sort out what it needs.

File: addons/microsoft_calendar/models/calendar.py
Comment (reth-odoo) [inline addons/microsoft_calendar/models/calendar.py:90]: As I remember this is because microsoft requires it, is it really ok to skip it merely because it's an appointment on our side?

If so great but it feels like it might be necessary to have the organizer as attendee in appointment,.

Comment (ryce-odoo) [reply addons/microsoft_calendar/models/calendar.py:90]: Hey @reth-odoo I don't know that it matters because from my testing on older code without this check in place, the calendar would just not sync that event if the organizer was not an attendee. Which should be fine in this case. I have tested this on old code and this is how the functionality worked before this commit https://github.com/odoo/odoo/commit/a2e823858bbd6d5199d71ad40747d6e633dd98fc

I am not sure if this is the proper functionality, but it seems better than giving customers unwanted emails when their customers book a resource.

File: addons/microsoft_calendar/models/calendar.py
Comment (ryce-odoo) [reply addons/microsoft_calendar/models/calendar.py:115]: Thanks for checking! Oversight on my part. I wrote this declaration before writing the appointment side and realizing I would need something from vals.

File: addons/microsoft_calendar/models/calendar.py
Comment (reth-odoo) [reply addons/microsoft_calendar/models/calendar.py:90]: Indeed, seems ok if it's handles it fine :)

Comment (reth-odoo) [reply addons/microsoft_calendar/models/calendar.py:90]: Note that in https://github.com/odoo/odoo/pull/129017 we expect that when syncing, all events will be synced and send an email to the participants. So if it's not the case here we'll need to adapt the logic a bit.

Maybe just updating `_check_microsoft_sync_status` to specify we won't sync for this case

Comment (ryce-odoo) [reply addons/microsoft_calendar/models/calendar.py:90]: Thanks for the heads up @reth-odoo. Haven't had a chance to work on this for a bit, but I am making some changes now.

**Comment by ryce-odoo** [comment]
Hey @reth-odoo 

I have made the changes to use a filter on the check function and I have tested it and everything seems to work. I am unsure why runbot is failing some JS tests because of this change though.

**Comment by ryce-odoo** [comment]
Hey @reth-odoo 

Can I get an approval on this PR and its related one if everything looks good? https://github.com/odoo/odoo/pull/162927

File: addons/microsoft_calendar/models/calendar.py
Comment (reth-odoo) [inline addons/microsoft_calendar/models/calendar.py:85]: More explicit names are better, also might be easier to filter values directly instead of zipping with booleans
```suggestion
        vals_check_organizer = self._check_organizer_validation_conditions(vals_list)
        for vals in [vals for vals, check_organizer in zip(vals_list, vals_check_organizer) if check_organizer]:
```

**Review by reth-odoo — COMMENTED** [review]
A few code organization comments on both PRs but otherwise lgtm
Don't forget to do the adjustments to send emails in 17.3!

sidenote: don't hesitate to ping me (on discord too), usually if I haven't responded in a couple of days it probably means I've forgotten about it

**Comment by ryce-odoo** [comment]
Hey @reth-odoo 

I committed your suggestions and made a change to the test to test the case in which an appointment should validate. The tests look all green so it should be good to go!

**Comment by reth-odoo** [comment]
:eyes: @ryce-odoo Hi, sorry but I don't see a 17.3 PR

Can't merge this without it as it will break the notification flow in stable otherwise.
Feel free to DM me if the required change is unclear

**Comment by ryce-odoo** [comment]
Hey @reth-odoo 

Sorry I haven't gotten a chance to getting around to working on this yet! I think my message was from before your most recent review! I will get around to this ASAP

**Comment by ryce-odoo** [comment]
Hey @reth-odoo 

Just bumping this I did dm you on Discord a while back asking a question

**Review by reth-odoo — APPROVED** [review]
Seems ok to me for now, we'll look into removing organizers from resource appointments as they don't make that much function change.

For now this works, if users configure it correctly it will sync as expected and if they don't (as is done automatically) we fail silently.

**Comment by reth-odoo** [comment]
@ryce-odoo don't forget to set forward port up to the version before https://github.com/odoo/odoo/pull/178142

**Comment by ryce-odoo** [comment]
Hey @reth-odoo 

Can you r+ the prs and set fw up to? I don't think I have the permission to unless you delegate.

**Comment by reth-odoo** [comment]
You should have the right to set fw up to without r+

I don't have r+ rights anyway, no :/

**Comment by arj-odoo** [comment]
@robodoo r+

**Comment by ryce-odoo** [comment]
@arj-odoo Can you r+ the related PR over here as well? 
https://github.com/odoo/enterprise/pull/61731

**Pull request closed without merging**
